### DISCUSSION TITLE
Singleton based on correlationId

### DISCUSSION THREAD

**Discussion by SamueleFurnari** [elsa-workflows/elsa-core#1507] [Q&A] [answered]
Hi guys,

As reported in the documentation, I correlate a domain object to the workflow using the object's Id as a correlation.

My need is to have a single running instance of the workflow based on the correlationId.

I'm searching a way to check if already exists a running workflow starting from the correlationId.

**Comment by sfmskywalker** [comment, marked as answer]
Hi @SamueleFurnari,

One easy way to see if there's already an active (non-completed, non-faulted) workflow is to use the `IWorkflowInstanceStore` and count the number of workflow instances for a given workflow definition ID and correlation ID, like this:

```csharp
var workflowDefinitionId= "some workflow definition ID";
var correlationId = "some correlation ID";

// Setup query specification.
var specification = new WorkflowDefinitionIdSpecification(workflowDefinitionId).And(new WorkflowIsAlreadyExecutingSpecification()).WithCorrelationId(correlationId);

// Count matching results.
var workflowInstanceExists = await _workflowInstanceStore.CountAsync(specification) > 0;
```

**Reply by SamueleFurnari** [reply]
Hi @sfmskywalker,

great! This works fine for my purpose! 
Thank you for the help
